modules/monitoring/library/Monitoring/Backend/Ido/Query/HoststatusQuery.php

```diff
@@ -117,35 +117,25 @@ class HoststatusQuery extends IdoQuery
             'host_process_performance_data'         => 'hs.process_performance_data',
             'host_retry_check_interval'             => 'hs.retry_check_interval',
             'host_scheduled_downtime_depth'         => 'hs.scheduled_downtime_depth',
-            'host_severity'                         => 'CASE WHEN hs.current_state = 0
-            THEN
-                CASE WHEN hs.has_been_checked = 0 OR hs.has_been_checked IS NULL
-                     THEN 16
-                     ELSE 0
-                END
-                +
-                CASE WHEN hs.problem_has_been_acknowledged = 1
-                     THEN 2
-                     ELSE
-                        CASE WHEN hs.scheduled_downtime_depth > 0
-                            THEN 1
-                            ELSE 4
-                        END
-                END
+            'host_severity'                         => 'CASE
+            WHEN hs.current_state = 0
+                THEN 1
+            WHEN hs.has_been_checked = 0 OR hs.has_been_checked IS NULL
+                THEN 16
             ELSE
-                CASE WHEN hs.has_been_checked = 0 OR hs.has_been_checked IS NULL THEN 16
-                     WHEN hs.current_state = 1 THEN 64
-                     WHEN hs.current_state = 2 THEN 32
-                     ELSE 256
+                CASE
+                WHEN hs.current_state = 1 THEN 64
+                WHEN hs.current_state = 2 THEN 32
+                ELSE 254
                 END
                 +
-                CASE WHEN hs.problem_has_been_acknowledged = 1
-                     THEN 2
-                     ELSE
-                        CASE WHEN hs.scheduled_downtime_depth > 0
-                            THEN 1
-                            ELSE 4
-                        END
+                CASE 
+                WHEN hs.problem_has_been_acknowledged = 1
+                    THEN 8
+                WHEN hs.scheduled_downtime_depth > 0
+                    THEN 4
+                WHEN (NOT hs.problem_has_been_acknowledged = 1) AND (NOT hs.scheduled_downtime_depth > 0)
```

Comment:
Why not use the CASE..ELSE structure from the previous code? In my opinion its more readable.